python/ray/_private/async_utils.py

```diff
@@ -0,0 +1,82 @@
+# Adapted from [aiodebug](https://gitlab.com/quantlane/libs/aiodebug)
+
+# Copyright 2016-2022 Quantlane s.r.o.
+
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+
+#        http://www.apache.org/licenses/LICENSE-2.0
+
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+# Modifications:
+# - Removed the dependency to `logwood`.
+# - Ranamed `log_slow_callbacks.enable()` to just `enable_log_slow_callbacks()`.
+# - Renamed `monitor_loop_lag.enable()` to just `enable_monitor_loop_lag()`.
+# - Miscellaneous changes to make it work with Ray.
+
+from asyncio.base_events import _format_handle
+from typing import Callable
+import asyncio
+import asyncio.events
+import time
+
+
+def enable_log_slow_callbacks(
+    slow_duration: float, on_slow_callback: Callable[[str, float], None] = None
+) -> None:
+    """
+    Patch ``asyncio.events.Handle`` to log warnings every time a callback takes
+    ``slow_duration`` seconds or more to run.
+
+    Note: this only works with asyncio's default event loop, not with uvloop.
+
+    :param on_slow_callback: Receives a formatted name of a slow callback
+        and the time (in seconds) it took to execute.
+    """
+    if on_slow_callback is None:
+        raise ValueError("on_slow_callback is required")
+    _run = asyncio.events.Handle._run
```

Comment:
so we have to rely on the private implementation details here?

Reply:
yes, this is private implementation details of asyncio.

Reply:
Added a message for it